tests/test_facebook_discovery.py

```diff
@@ -0,0 +1,154 @@
+                        msg="One or more streams don't follow standard naming")
+
+        for stream in streams_to_test:
+            with self.subTest(stream=stream):
+                catalog = next(iter([catalog for catalog in found_catalogs
+                                     if catalog["stream_name"] == stream]))
+                assert catalog  # based on previous tests this should always be found
+
+                schema_and_metadata = menagerie.get_annotated_schema(conn_id, catalog['stream_id'])
+                metadata = schema_and_metadata["metadata"]
+                schema = schema_and_metadata["annotated-schema"]
+
+                # verify there is only 1 top level breadcrumb
+                stream_properties = [item for item in metadata if item.get("breadcrumb") == []]
+                self.assertTrue(len(stream_properties) == 1,
+                                msg="There is NOT only one top level breadcrumb for {}".format(stream) + \
+                                "\nstream_properties | {}".format(stream_properties))
+
+                # BUG_1 | https://stitchdata.atlassian.net/browse/SRCE-4855
+                failing_with_no_replication_keys = {
+                    'ads_insights_country', 'adsets', 'adcreative', 'ads', 'ads_insights_region',
+                    'campaigns', 'ads_insights_age_and_gender', 'ads_insights_platform_and_device',
+                    'ads_insights_dma', 'ads_insights'
+                }
+                if stream not in failing_with_no_replication_keys:  # BUG_1
+                    # verify replication key(s)
+                    self.assertEqual(
+                        set(stream_properties[0].get(
+                            "metadata", {self.REPLICATION_KEYS: []}).get(self.REPLICATION_KEYS, [])),
+                        self.expected_replication_keys()[stream],
+                        msg="expected replication key {} but actual is {}".format(
+                            self.expected_replication_keys()[stream],
+                            set(stream_properties[0].get(
+                                "metadata", {self.REPLICATION_KEYS: None}).get(
+                                    self.REPLICATION_KEYS, []))))
+
+                # verify primary key(s)
+                self.assertEqual(
+                    set(stream_properties[0].get(
+                        "metadata", {self.PRIMARY_KEYS: []}).get(self.PRIMARY_KEYS, [])),
+                    self.expected_primary_keys()[stream],
+                    msg="expected primary key {} but actual is {}".format(
+                        self.expected_primary_keys()[stream],
+                        set(stream_properties[0].get(
+                            "metadata", {self.PRIMARY_KEYS: None}).get(self.PRIMARY_KEYS, []))))
+
+                actual_replication_method = stream_properties[0].get(
+                    "metadata", {self.REPLICATION_METHOD: None}).get(self.REPLICATION_METHOD)
```

Comment:
Is this on two lines because it's too long? If so, `.get("metadata", {})` would get you the same output and it might be small enough to fit on one line

Reply:
Done